index.html

```diff
@@ -3400,33 +3418,24 @@ <h4>Convert to Absolute URL</h4>
 							<li id="abs-url-convert">
 								<p>if <var>value</var> is a <a href="https://infra.spec.whatwg.org/#string">string</a>,
 									set <var>value</var> to the result of running the <a
-										href="https://url.spec.whatwg.org/#concept-url-parser">URL parser</a> with
-										<var>value</var> as input and <var>base</var> as the base URL.</p>
-							</li>
-							<li id="abs-url-recurse">
-								<p>otherwise, if <var>value</var> is a <a href="https://infra.spec.whatwg.org/#list"
-										>list</a>, <a href="https://infra.spec.whatwg.org/#list-iterate">for each</a>
-									<var>item</var> of <var>value</var>, set <var>item</var> to the result of runing
-										<a>convert to absolute URL</a>, when successful, given <var>item</var> and
-										<var>base</var>. Otherwise, <a href="https://infra.spec.whatwg.org/#list-remove"
-										>remove</a>
-									<var>item</var> from <var>value</var>.</p>
+										href="https://url.spec.whatwg.org/#concept-url-parser">URL
+									parser</a>&#160;[[!url]], when successful, with <var>value</var> as input and
+										<var>base</var> as the base URL. Otherwise, <a>validation error</a>, return
+									failure.</p>
 							</li>
 							<li id="abs-url-error">
 								<p>otherwise, <a>validation error</a>, return failure.</p>
 							</li>
 							<li id="abs-url-return">
-								<p>Return <var>value</var>.</p>
+								<p>return <var>value</var>.</p>
 							</li>
 						</ol>
 						<details>
 							<summary>Explanation</summary>
 							<p>This function checks if the value sent into it is a relative URL, and, if so, converts it
-								to an absolute URL. If the value is a <a href="https://infra.spec.whatwg.org/#list"
-									>list</a>, each item in the list is recursively passed to this function to inspect
-								it.</p>
-							<p>If the value passed to the function is not a string or a list, a failure is returned.
-								Otherwise, the updated value is returned.</p>
+								to an absolute URL.</p>
```

Comment:
To be checked... but I believe the URL parser that we refer to here does not check whether the resulting URL is an https or http URL. On the other hand, our spec says that we restrict these URLs to those two. 

If the referred WHATWG function indeed does not check this, then our function must have an addition check on the scheme. If it does, it is worth emphasizing this in the explanation.

Reply:
Right, the url parser doesn't require a specific scheme. I'll add an extra step into the function to also check the scheme of the returned url.

Reply:
I did a couple of tweaks to this section in the last commit. In addition to adding the scheme step, I also put a first step in to check that both the incoming value and base are non-empty strings.

I don't think it's problematic to require base be a non-empty string, since manifests have to have a base. If they don't, the publication is invalid, although this step will have the unintended consequence of removing all URLs in such a scenario, even if they are absolute already.